### PR TITLE
EnC - Support record structs

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/51200

Thank you @jcouv for making this really straightforward 😀

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs [hunk 1]
```diff
@@ -1693,6 +1693,17 @@ public void Record_Name_Update()
                 Diagnostic(RudeEditKind.Renamed, "record D", CSharpFeaturesResources.record_));
         }
 
+                [Fact]
```
Comment (davidwengier) [inline src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs:1696]:
```suggestion
        [Fact]
```

**Comment by Youssef1313** [comment]
> Fixes #44877

This issue is already closed (it was for records) - probably there is another issue for record structs?

**Comment by davidwengier** [comment]
> This issue is already closed (it was for records) - probably there is another issue for record structs?

Thanks, bad copy-paste :)

File: src/Features/CSharp/Portable/CSharpFeaturesResources.resx [hunk 2]
```diff
@@ -621,4 +621,7 @@
   <data name="record_" xml:space="preserve">
     <value>record</value>
   </data>
+  <data name="recordstruct" xml:space="preserve">
```
Comment (Youssef1313) [inline src/Features/CSharp/Portable/CSharpFeaturesResources.resx:624]: nit: the naming is usually split by `_`

**Comment by tmat** [comment]
We should test that adding fields to record struct reports `RudeEditKind.InsertIntoStruct` rude edit.

**Comment by tmat** [comment]
Are there any differences in synthesized members between records and record structs?

**Comment by davidwengier** [comment]
> Are there any differences in synthesized members between records and record structs?

PrintMembers has different visibility, and there is no `Clone$` method or copy constructor, but nothing matters for EnC. If fields were able to be added to structs then we'd at least want to add a test to ensure the semantic edits are right due to the missing copy constructor, but based on how the code is written (ie it enumerates constructors rather than assuming they exist) I don't think it would be an issue.

**Comment by davidwengier** [comment]
@tmat is there anything else for this? any other tests you think are important?

**Review by tmat — APPROVED** [review]
:shipit:

**Comment by tmat** [comment]
One more thing we might want to test is changing `record` to `record struct` and vice versa. I'm guessing we already correctly report rude edit.

**Comment by davidwengier** [comment]
We do, but I'll add a test anyway.

**Comment by jcouv** [comment]
Thanks!

**Pull request merged**